### PR TITLE
set custom limits on memory through command line flag

### PR DESCRIPTION
**Pull request by yshen-google**
Issue #2047
Allow user to set custom memory amount through flag and taking the minimum of the custom memory and system total virtual memory.
Tested locally through flag control and memory amount was returned correctly.
@percyliang

### CODE REVIEW

File: codalab/worker/worker.py [hunk 1]
```diff
@@ -50,6 +50,7 @@ def __init__(
         commit_file,  # type: str
         cpuset,  # type: Set[str]
         gpuset,  # type: Set[str]
+        max_memory,  # type:Optinal[int]
```
Comment (nelson-liu) [inline codalab/worker/worker.py]: `s/Optinal/Optional/`

File: codalab/worker/worker.py [hunk 2]
```diff
@@ -50,6 +50,7 @@ def __init__(
         commit_file,  # type: str
         cpuset,  # type: Set[str]
         gpuset,  # type: Set[str]
+        max_memory,  # type:Optional[int]
```
Comment (candicegjing) [inline codalab/worker/worker.py]: Add a space between `:` and `Optional` and remove the square bracket around `int`.

**Review by candicegjing — COMMENTED** [review]
Thanks for doing this! Other than the small formatting issue, everything else looks great to me.

File: codalab/worker/worker.py (hunk 2 above)
Comment (yshen-google) [reply codalab/worker/worker.py]: Thanks for the review.
Space is added.
Regarding square bracket, it seems the other occurrences also use `Optional[{data type}]` style such as `dependency_manager,  # type: Optional[DependencyManager]` and `local_bundles_dir,  # type: Optional[str]`.

**Review by percyliang — APPROVED** [review]
Thanks!

File: codalab/worker/worker.py (hunk 2 above)
Comment (candicegjing) [reply codalab/worker/worker.py]: Yes, you are right. Annotated arguments with a default value of None should use `[]`.

**Review by candicegjing — APPROVED** [review]
Thanks for doing this! Looks good to me!

**Pull request merged**